src/Extensions/Annotatable.php

```diff
@@ -8,68 +14,69 @@
  *
  * @package IDEAnnotator/Extensions
  *
- * @property DevBuildController $owner
+ * @property \SilverStripe\Dev\DevBuildController $owner
  */
 class Annotatable extends Extension
 {
 
     /**
-     * @var DataObjectAnnotator
-     */
-    protected $annotator;
-
-    /**
-     * @var AnnotatePermissionChecker
-     */
-    protected $permissionChecker;
-
-    /**
      * Keep track ot the annotation actions for extensions
      * An Extension can belong to many DataObjects.
      * This prevents that an Extension is ran twice on dev/build
      * @var array
      */
     public static $annotated_extensions = array();
-
     /**
-     * Annotatable setup.
-     * This is theoratically a constructor, but to save memory we're using setup called from {@see requireDefaultRecords}
+     * @var DataObjectAnnotator
      */
-    public function setUp()
-    {
-        $this->annotator = Injector::inst()->get('DataObjectAnnotator');
-        $this->permissionChecker = Injector::inst()->get('AnnotatePermissionChecker');
-    }
+    protected $annotator;
+    /**
+     * @var AnnotatePermissionChecker
+     */
+    protected $permissionChecker;
 
     /**
      * Annotated Controllers and Extensions
+     * @throws \Psr\Container\NotFoundExceptionInterface
      */
     public function afterCallActionHandler()
     {
         $this->setUp();
 
         $skipAnnotation = $this->owner->getRequest()->getVar('skipannotation');
-        $envIsAllowed   = $this->permissionChecker->environmentIsAllowed();
-        
+        $envIsAllowed = Director::get_environment_type() === 'dev';
```

Comment:
It might be best to use `Director::isDev()` so we remove the "dev" string which could change